contrib/scripts/check_oid_ovfps.sh

```diff
@@ -0,0 +1,515 @@
+
+	if [ $? -ne 0 ];then
+		passwd=$(get_password)
+		pass="-p $passwd"
+
+		passwd=$(csql ${connection_mode} ${user} ${pass} -c "SELECT 1" $database 2> /dev/null)
+		if [ $? -ne 0 ];then
+			echo "$dbuser: Incorrect or missing password"
+			exit 1
+		fi
+	fi
+
+	# check whether this user is a member of DBA groups
+	#
+	dba_groups=$(csql  ${connection_mode}  ${user} ${pass} -l -c "$qry_dba_grp" $database | grep -w $USERNAME | wc -l)
+	if [ $USERNAME != "DBA" ] && [ $dba_groups -eq 0 ];then
+		echo "User '$dbuser' is not a member of DBA group"
+		exit 2
+	fi
+}
+
+# exit, if subject db server is not running
+function check_db_server_running ()
+{
+	local db=$1
+	local mode=$2
+
+	db=${database%%@*}
+
+	retcode=$(ps -ef | grep cub_server | grep $db | wc -l)
+	if [ $retcode -eq 0 ];then
+		if [ x"${mode}" == x"-C" ]; then
+			echo "Database server '$database' is not running"
+			exit 1
+		fi
+	else
+		if [ x"${mode}" != x"-C" ]; then
+			echo "Database server '$database' is running"
+			exit 1
+		fi
+	fi
+}
+
+#
+# when doing a cd, cd display current working directory.
+# skip it.
+#
+function silent_cd ()
+{
+	cd $* > /dev/null
+}
+
+
+function IFS_cleanup() 
+{       
+        IFS=
+}
+
+function IFS_restore()
+{
+        IFS=${IFS_backup}
+}
+
+function do_get_index_name_list()
+{
+	local is_title_line=1
+	local res_fnm="ti.res"
+	local csql_options=""
+	local i_qry=""
+	local csql_qry=""
+
+	i_qry="${i_qry} SELECT CAST (c.owner.name AS VARCHAR(255)) AS owner_name,"
+	i_qry="${i_qry}         c.class_name AS class_name,"
+	i_qry="${i_qry}         i.index_name AS index_name,"
+	i_qry="${i_qry}         NVL( (SELECT CASE WHEN pname IS NULL THEN 'P' ELSE 'S' END"
+	i_qry="${i_qry}               FROM _db_partition p WHERE p.class_of =c), 'X') AS partitioned,"
+	i_qry="${i_qry}         CASE i.status WHEN 2 THEN 'YES' ELSE 'NO' END AS invisible"
+	i_qry="${i_qry} FROM _db_index i, _db_class c"
+	i_qry="${i_qry} WHERE i.class_of = c"
+	i_qry="${i_qry}       AND c.class_type=0 AND (MOD(c.is_system_class, 2) != 1)"
+	i_qry="${i_qry}       AND i.is_unique=0  AND (i.status=1 /* NORMAL */ OR i.status=2 /* INVISIBLE */)"
+
+	csql_qry="SELECT owner_name || '.' || class_name || '.' ||  index_name || '.' || partitioned || '.' || invisible  "
+	csql_qry="${csql_qry} FROM ( ${i_qry} ) AS t"
+	csql_qry="${csql_qry} WHERE 1=1"
+
+	if [ x"${target_owner}" != x"" ]; then
+		# For comparison of owner_name, change it to uppercase and enter it.
+		csql_qry="${csql_qry} AND t.owner_name='${target_owner^^}'"
+	fi
+
+	if [ x"${target_table}" != x"" ]; then
+		csql_qry="${csql_qry} AND ( t.class_name='${target_table}'"
+		csql_qry="${csql_qry}        OR (t.class_name LIKE '${target_table}__p__%' AND t.partitioned = 'S'))"
+	fi
+
+	csql_qry="${csql_qry} ORDER BY owner_name, class_name, partitioned, index_name"
+
+    # send query
+	csql_options="${connection_mode} ${user} ${pass} -t"
+	csql ${csql_options} -c "${csql_qry}" $database > "${res_fnm}"
```

Comment:
```suggestion
	S=$(csql ${csql_options} -c "${csql_qry}" $database)
```
and at line 310
```
       done <<< ${S}
```
if you want to avoid saving the intermediate result into a file.

Reply:
Thank you
 I will fix it.